### PR TITLE
Jlopezbarb/fix last updated not being updated

### PR DESCRIPTION
**Pull request by jLopezbarb**
# Proposed changes

- Add a comment on why we use a different client for configmap
- Add the `lastUpdated` annotation for the configmap created on `okteto deploy`

### CODE REVIEW

**Review by teresaromero — APPROVED** [review]

File: cmd/deploy/command_configuration_test.go [hunk 1]
```diff
@@ -92,7 +92,11 @@ devs:
 		t.Fatal("error trying to get configmap from data object")
 	}
 
-	assert.Equal(t, expectedCfg, currentCfg)
+	assert.Equal(t, expectedCfg.Name, currentCfg.Name)
+	assert.Equal(t, expectedCfg.Namespace, currentCfg.Namespace)
+	assert.Equal(t, expectedCfg.Labels, currentCfg.Labels)
+	assert.Equal(t, expectedCfg.Data, currentCfg.Data)
+	assert.NotNil(t, currentCfg.Annotations[model.LastUpdatedAnnotation])
```
Comment (ifbyol) [inline cmd/deploy/command_configuration_test.go:99]: If I'm not wrong, `currentCfg.Annotations[model.LastUpdatedAnnotation]` cannot be nil as Annotations is a `map[string]string` and the default value would be an empty string. So this assert would always be true even if the annotation is not present.

There would be 2 ways of checking this:
* `_, ok := currentCfg.Annotations[model.LastUpdatedAnnotation]` and check that `ok` is true
* `value := currentCfg.Annotations[model.LastUpdatedAnnotation]` and verify that `val != ""`

Same about the tests in `deploy_test.go` and `translate_test.go`

File: pkg/cmd/pipeline/translate_test.go [hunk 2]
```diff
@@ -91,6 +93,9 @@ func Test_AddDevAnnotations(t *testing.T) {
 	manifest := &model.Manifest{
 		Namespace: "namespace",
 		Dev: model.ManifestDevs{
+			"not-found-deployment": &model.Dev{
+				Name: "not-found-deployment",
+			},
```
Comment (ifbyol) [inline pkg/cmd/pipeline/translate_test.go:98]: Just curiosity, Why this change?

**Review by ifbyol — APPROVED** [review]
Code looks good but take a look to the comment I added in the tests before merging please!

File: pkg/cmd/pipeline/translate_test.go (hunk 2 above)
Comment (jLopezbarb) [reply pkg/cmd/pipeline/translate_test.go:98]: That tests an untested scenario which checks what happens if the deployment is no longer running. We should skip the adding annotation function for that resource
